### DISCUSSION TITLE
using same URL variable as binding to lead to different screens

### DISCUSSION THREAD

**Discussion by moosri** [Budibase/budibase#8196] [Help] [answered]
Hello, I am pretty new to budibase and have a question regarding the binding and linking of URL variables...

I have a starting screen displaying a table, then created two different screens that should show a row entry of this table in a new screen, once as write-only and once as read-only with different information. They should both be linked via the id.
The read-only screen is working when I simply use linking rows and then choose the /:id variable with a {{URL.id}}-binding, but when I try to create a new button into my start screen which should navigate to the write-only screen using a new path named /:id_edit (which should also use the id binding to display the row entry) it ignores the screen linked via "linking-table-rows".

I have read something about a bug regarding this but I am unsure whether it is just me or a bug.

Ty!

**Comment by melohagan** [comment]
Hey @moosri 

I think I understand what you mean - it may be the same scenario that I came across here: https://github.com/Budibase/budibase/pull/8016#issuecomment-1261929432

Still I think it would be good to see a few screenshots or a video to demonstrate what the issue is!

**Reply by moosri** [reply]
Hey @melohagan 

I had read through the #8016 comment but it is sadly not what I meant. 

I attached some photos to explain my problem. 
On my home screen, I displayed a table called 'marketplaces'. I want to link the table rows to another screen that displays each detailed view of a marketplace. I did this via URL-path-binding and used an attribute called id. Abstractly, the detailed view to which this linking should lead should be a "read-only" view on this detailed marketplace. Now I want to link to another second screen via a "view-link" on my home screen which has the same structure as my detailed marketplace view and also uses the {{URL.id}} path-variable as a binding. Sadly it is not working because it is not linking the detailed view with the same Id. 


Thank you very very much for your help, I am very new to budibase and I am a bit lost. 

<img width="827" alt="marketplaces_edit_id_screen" src="https://user-images.githubusercontent.com/104364585/196156564-4fffdcf8-4742-4584-988b-8535fcf3748b.PNG">
<img width="816" alt="marketplaces_homescreen" src="https://user-images.githubusercontent.com/104364585/196156569-45c8dc3e-caf6-4aa2-ae70-63568791dff0.PNG">
<img width="810" alt="marketplaces_id_screen" src="https://user-images.githubusercontent.com/104364585/196156571-ae194355-d522-4092-ab14-7f74ff1ae481.PNG">
<img width="261" alt="screen_linking" src="https://user-images.githubusercontent.com/104364585/196156575-07c442b4-c5fc-4493-acc5-e14f90d5be5e.PNG">
<img width="577" alt="screennames" src="https://user-images.githubusercontent.com/104364585/196156576-95f9360c-a66f-497c-9403-4ebef72d5305.PNG">

**Comment by melohagan** [comment, marked as answer]
Hey @moosri 

Thanks for the screenshots!

So the issue here is that there is a difference in how the `Link screen` for the `Link table rows` property works and how links work.

Basically your **Edit** link needs to reference the URL binding, i.e. `/marketplaces/edit/{{ URL.id }}`

Whereas the link table rows should be fine as is. This page should be of some use: https://docs.budibase.com/docs/url-parameters

**Reply by moosri** [reply]
Dear @melohagan 

It works now, thank you very much!

**Reply by melohagan** [reply]
Awesome, glad to hear it!
